src/analysis/sampling/hit_and_run.jl

```diff
@@ -47,33 +47,70 @@ samples = hit_and_run(100_000, opt_model; keepevery=10, samplesize=5000)
 ```
 """
 function hit_and_run(
-    N::Int,
-    opt_model;
+    model,
+    optimizer;
+    modifications = [],
+    N = 1000,
     keepevery = _constants.sampling_keep_iters,
     samplesize = _constants.sampling_size,
-    random_objective = false,
+    warmup_indices = collect(1:n_reactions(model)),
+    workers = [myid()],
+    nchains = 1,
 )
 
-    lbs, ubs = get_bound_vectors(opt_model) # get actual ub and lb constraints, can't use model function because the user may have changed them in the function arguments
+    # get warmup points in parallel, similar to FVA
+    ws, lbs, ubs = warmup(
+        model,
+        optimizer;
+        modifications = modifications,
+        warmup_points = warmup_indices,
+        workers = workers, # parallel
+    )
+
+    # load warmup points to workers
+    map(fetch, save_at.(workers, :ws, Ref(:($ws))))
+    map(fetch, save_at.(workers, :lbs, Ref(:($lbs))))
+    map(fetch, save_at.(workers, :ubs, Ref(:($ubs))))
+
+    # do in parallel! 
+    samples = dpmap(
+        x -> :($COBREXA._serial_hit_and_run(ws, lbs, ubs, $samplesize, $keepevery, $N)),
+        CachingPool(workers),
+        1:nchains,
+    )
 
-    wpoints = _get_warmup_points(opt_model; random_objective = random_objective)
+    # remove warmup points from workers
+    map(fetch, remove_from.(workers, :ws))
+    map(fetch, remove_from.(workers, :lbs))
+    map(fetch, remove_from.(workers, :ubs))
+
+    # not sure how to do this better
+    vals = zeros(samplesize, length(lbs), nchains)
```

Comment:
I guess something as in `vcat(samples...)` could help, e.g. like `cat(dim=3, samples...)`

Reply:
didn't work, no biggie though :)